### PR TITLE
Add a git_stash rule

### PR DESCRIPTION
**Pull request by mcarton**

### CODE REVIEW

**Comment by nvbn** [comment]
Thanks!

File: thefuck/rules/git_stash.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def match(command, settings):
+    return ('git' in command.script
+            and 'Please commit or stash them.' in command.stderr)
+
+
+def get_new_command(command, settings):
+    return 'git stash && ' + command.script
```
Comment (nwinkler) [inline thefuck/rules/git_stash.py]: Shouldn't it unstash the changes after running the command? Like `git stash pop`?

File: thefuck/rules/git_stash.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def match(command, settings):
+    return ('git' in command.script
+            and 'Please commit or stash them.' in command.stderr)
```
Comment (dunn) [inline thefuck/rules/git_stash.py]: Could this be tweaked to work for switching branches too?  That error message is `Please, commit your changes or stash them before you can switch branches.` So maybe the `match` could be as simple as `or stash them`.

**Comment by mcarton** [comment]
@nicwest: well, I don't like `git stash pop`; see https://codingkilledthecat.wordpress.com/2012/04/27/git-stash-pop-considered-harmful/ and plenty others.
@dunn: sure, I will do that tomorrow.

**Comment by nicwest** [comment]
@nwinkler: I think he meant that for you.

**Pull request merged**